extensions/Worker.Extensions.CosmosDB/src/CosmosDBConverter.cs

```diff
@@ -148,11 +144,11 @@ private async Task<object> CreatePOCOCollectionAsync<T>(Container container, Cos
             QueryRequestOptions queryRequestOptions = new();
             if (!String.IsNullOrEmpty(cosmosAttribute.PartitionKey))
             {
-                var partitionKey =  new PartitionKey(cosmosAttribute.PartitionKey);
+                var partitionKey = new PartitionKey(cosmosAttribute.PartitionKey);
                 queryRequestOptions = new() { PartitionKey = partitionKey };
             }
 
-            using (var iterator = container.GetItemQueryIterator<T>(queryDefinition: queryDefinition, requestOptions: queryRequestOptions))
+            using (var iterator = container.GetItemQueryIterator<dynamic>(queryDefinition: queryDefinition, requestOptions: queryRequestOptions))
```

Comment:
Similarly here, can use `GetItemQueryStreamIterator`

Reply:
Switched to stream by I'm still doing serialize to json string -> then deserialize to target type via the "ConvertObject" method. I don't really love this and was wondering if it would be better to deserialize every document to target type when building the list i.e.

```csharp
        private async Task<IList<object>> ExtractCosmosDocumentsAsync(FeedIterator iterator)
        {
            var documentList = new List<Stream>();
            while (iterator.HasMoreResults)
            {
                ResponseMessage response = await iterator.ReadNextAsync();
                var doc = return JsonSerializer.Deserialize(response.Content, targetType);
                documentList.Add(doc);
            }
            return documentList;
        }
``` 

But the serialize/deserialize approach felt less heavy?

Reply:
I would look into using an `IAsyncEnumerable<Stream>` with the `yield` keyword. This would let you avoid have to realize the entire query right away into a `List`.